File: examples/reactNative-demo/android/app/src/main/java/com/reactnativedemo/plaud/PlaudReactPackage.kt
package com.reactnativedemo.plaud

import com.facebook.react.ReactPackage
import com.facebook.react.bridge.NativeModule
import com.facebook.react.bridge.ReactApplicationContext
import com.facebook.react.uimanager.ViewManager

class PlaudReactPackage : ReactPackage {
    override fun createNativeModules(reactContext: ReactApplicationContext): List<NativeModule> {
        return listOf(
            PlaudSDKModule(reactContext),
            PlaudBluetoothModule(reactContext),
            PlaudRecordingModule(reactContext),
            PlaudFileManagerModule(reactContext),
            PlaudUploadModule(reactContext),
            PlaudPermissionModule(reactContext),
            PlaudEnvironmentModule(reactContext)
        )
    }

    override fun createViewManagers(reactContext: ReactApplicationContext): List<ViewManager<*, *>> {
        return emptyList()
    }
}

File: examples/reactNative-demo/ios/PlaudSDK/PlaudReactPackage.m
//
//  PlaudReactPackage.m
//  ReactNativeDemo
//
//  Created by Plaud Team on 2024/12/19.
//

#import "PlaudReactPackage.h"
#import "PlaudSDKModule.h"
#import "PlaudBluetoothModule.h"
#import "PlaudRecordingModule.h" 
#import "PlaudFileManagerModule.h"
#import "PlaudUploadModule.h"
#import "PlaudPermissionModule.h"
#import "PlaudEnvironmentModule.h"

@implementation PlaudReactPackage

RCT_EXPORT_MODULE();

+ (NSArray<id<RCTBridgeModule>> *)plaudModules
{
    return @[
        [[PlaudSDKModule alloc] init],
        [[PlaudBluetoothModule alloc] init],
        [[PlaudRecordingModule alloc] init],
        [[PlaudFileManagerModule alloc] init],
        [[PlaudUploadModule alloc] init],
        [[PlaudPermissionModule alloc] init],
        [[PlaudEnvironmentModule alloc] init]
    ];
}

@end
